pkg/mermaid/mermaidsequencediagram.go

```diff
@@ -0,0 +1,157 @@
+			if isloop || iselse {
+				result += fmt.Sprintf("%send\n", addIndent(indent))
+			}
+		case *sysl.Statement_Call:
+			nextapp := c.Call.Target.Part[0]
+			nextep := c.Call.Endpoint
+			pair := sequencepair{appname, nextep, nextep}
+			if !sequencenpairscontain(*sequencepairs, pair) {
+				*sequencepairs = append(*sequencepairs, pair)
+				result += callStatement(appname, nextep, nextapp, indent)
+				previous := appname
+				out, err := generateMermaidSequenceDiagramHelper(m, nextapp, nextep, previous, indent, sequencepairs, false)
+				if err != nil {
+					panic("Error in generating sequence diagram; check if app names or endpoints are correct")
+				}
+				result += out
+			}
+		case *sysl.Statement_Ret:
+			retep := c.Ret.Payload
+			result += retStatement(appname, retep, previousapp, indent, thestart)
+		case *sysl.Statement_Action:
+			result += actionStatement(appname, c.Action.Action, indent)
+		case *sysl.Statement_Cond:
+			result += fmt.Sprintf("%salt %s\n", addIndent(indent), c.Cond.Test)
+			result += printSequenceDiagramStatements(m, c.Cond.Stmt, appname, previousapp, indent+1, sequencepairs, thestart)
+			if count+1 < len(statements) {
+				switch temp := statements[count+1].Stmt.(type) {
+				case *sysl.Statement_Group:
+					ok := startElse.MatchString(temp.Group.Title)
+					if !ok {
+						result += fmt.Sprintf("%send\n", addIndent(indent))
+					}
+				default:
+					result += fmt.Sprintf("%send\n", addIndent(indent))
+				}
+			} else {
+				result += fmt.Sprintf("%send\n", addIndent(indent))
+			}
+		case *sysl.Statement_Foreach:
+			result += fmt.Sprintf("%sloop %s\n", addIndent(indent), c.Foreach.Collection)
+			result += printSequenceDiagramStatements(m, c.Foreach.Stmt, appname, previousapp, indent+1, sequencepairs, thestart)
+			result += fmt.Sprintf("%send\n", addIndent(indent))
+		case *sysl.Statement_Loop:
+			result += fmt.Sprintf("%sloop %s\n", addIndent(indent), c.Loop.Criterion)
+			result += printSequenceDiagramStatements(m, c.Loop.Stmt, appname, previousapp, indent+1, sequencepairs, thestart)
+			result += fmt.Sprintf("%send\n", addIndent(indent))
+		case *sysl.Statement_LoopN:
+			result += fmt.Sprintf("%sloop %d times\n", addIndent(indent), c.LoopN.Count)
+			result += printSequenceDiagramStatements(m, c.LoopN.Stmt, appname, previousapp, indent+1, sequencepairs, thestart)
+			result += fmt.Sprintf("%send\n", addIndent(indent))
+		default:
+			panic("Unrecognised statement type")
+		}
+		count++
+	}
+	return result
+}
+
+func validAppNameAndEndpoint(m *sysl.Module, appname string, epname string) error {
+	if _, ok := m.Apps[appname]; !ok {
+		return errors.New("invalid app name")
+	}
+	if _, ok := m.Apps[appname].Endpoints[epname]; !ok {
+		return errors.New("invalid endpoint")
+	}
+	return nil
+}
+
+func callStatement(appname string, epname string, nextapp string, indent int) string {
+	var out = addIndent(indent)
+	out += fmt.Sprintf("%s->>+%s: %s\n", appname, nextapp, epname)
+	return out
```

Comment:
`return fmt.Sprintf("%s%s->>+%s: %s\n",addIndent(indent), appname, nextapp, epname)`

Reply:
Got it